extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorStrict.java

```diff
@@ -22,22 +22,28 @@ static <T> ObjectProcessor<T> create(ObjectProcessor<T> delegate) {
     }
 
     private final ObjectProcessor<T> delegate;
+    private final List<Type<?>> outputTypes;
 
     ObjectProcessorStrict(ObjectProcessor<T> delegate) {
         this.delegate = Objects.requireNonNull(delegate);
+        this.outputTypes = List.copyOf(delegate.outputTypes());
     }
 
     @Override
     public List<Type<?>> outputTypes() {
-        return delegate.outputTypes();
+        final List<Type<?>> outputTypes = delegate.outputTypes();
+        if (!this.outputTypes.equals(outputTypes)) {
+            throw new UncheckedDeephavenException("Implementation is returning a different list of outputTypes");
+        }
```

Comment:
Do you think this is actually a real worry?

Reply:
No, but the strict impl _is_ being a bit pedantic on purpose.